src/pkg/services/m365/api/graph/service.go

```diff
@@ -407,11 +411,18 @@ func (aw *adapterWrap) Send(
 		case IsErrBadJWTToken(err):
 			logger.Ctx(ictx).Debug("bad jwt token")
 			events.Inc(events.APICall, "badjwttoken")
+		case requestInfo.Method.String() == http.MethodGet && IsErrInvalidRequest(err):
+			// Graph may sometimes return a transient 400 response during onedrive
+			// and sharepoint backup. This is pending investigation on msft end, retry
+			// for now as it's a transient issue. Restrict retries to GET requests only
+			// to limit the scope of this fix.
+			logger.Ctx(ictx).Debug("invalid request")
+			events.Inc(events.APICall, "invalidrequest")
 		default:
 			return nil, clues.StackWC(ictx, err).WithTrace(1)
 		}
 
-		time.Sleep(3 * time.Second)
+		time.Sleep(adapterRetryDelay)
```

Comment:
could we actually make this a field on the adapterWrapper struct? That would allow for changing it during tests (tests can modify the instance of the struct prior to running tests), allow having a const for it, and ensure isolation between tests by ensuring changes from one test don't affect anything else

Reply:
Great point! Let me switch to this pattern.